### ISSUE TITLE
$open all

### ISSUE THREAD

**Issue by Zukunftsmusik** [Gelbpunkt/IdleRPG#158]
If I have a bunch of crates (say 12 after leveling up to lvl 12) I'd have to type $open 12 times into the chat. That's tiresome and boring. 

I'd like to have a `$open all` command to be able to open all crates at once. The answer from the bot should be much less detailed than usually, maybe just a list of the items found.

**Comment by maxx0911** [comment]
This could be problematic if someone had, say 500 crates. It clashes with the 2000 character limit per message and would probably slow down the bot too.
Instead of a command that opens all crates, there should be an amount of crates you want to open, ex `$open 15` where there can be a limit of how many crates you can open at once.

**Comment by Zukunftsmusik** [comment]
> This could be problematic if someone had, say 500 crates. It clashes with the 2000 character limit per message and would probably slow down the bot too.
> Instead of a command that opens all crates, there should be an amount of crates you want to open, ex `$open 15` where there can be a limit of how many crates you can open at once.

That's true, I haven't thought about that. Maybe a combination of both. The `$open all` command could be a short form for the current max amount of crates you're allowed to open at the same time. I'll edit the issue description.

**Comment by Quinemaniac** [comment]
We should make "$open all" orderly or it will be very spammy and kill the bot

**Comment by Zukunftsmusik** [comment]
> We should make "$open all" orderly or it will be very spammy and kill the bot

I don't get it. What do you mean?

**Comment by Quinemaniac** [comment]
> > We should make "$open all" orderly or it will be very spammy and kill the bot
> 
> I don't get it. What do you mean?

adrian already thought about it but maybe he stopped planning on it now.

**Comment by Zukunftsmusik** [comment]
> > > We should make "$open all" orderly or it will be very spammy and kill the bot
> > 
> > 
> > I don't get it. What do you mean?
> 
> adrian already thought about it but maybe he stopped planning on it now.

Having an open all command with only one output would be less spammy than opening a bunch of crates one by one. That's another reason why I suggested it. The output of such a command could follow the output of the `$i` command.

**The current output is**
```You gained an item!
You found a new item when opening a crate!
ID
<ID>
Name
<Name>
Type
<Type>
Damage
<Damage>
Armor
<Armor>
Value
<Value>```
times 15 when opening 15 boxes.

**The new output of e.g. `$open 5` could be**
```You gained 5 items!
<Name>
ID: <ID>, Type: <Type> with <EnhancementType>: <EnhancementValue>. Value is <Value>
<Name>
ID: <ID>, Type: <Type> with <EnhancementType>: <EnhancementValue>. Value is <Value>
<Name>
ID: <ID>, Type: <Type> with <EnhancementType>: <EnhancementValue>. Value is <Value>
<Name>
ID: <ID>, Type: <Type> with <EnhancementType>: <EnhancementValue>. Value is <Value>
<Name>
ID: <ID>, Type: <Type> with <EnhancementType>: <EnhancementValue>. Value is <Value>```
This would be much cleaner than before, much less spammy and with an appropriate cooldown it wouldn't even add more load on the bot's performance.


I also still don't understand what you mean by `[would] kill the bot.`
